### ISSUE TITLE
A typo or an explicit design in detect net's 5-dimension output?（大神求回复）

### ISSUE THREAD

**Issue by duducheng** [lfz/DSB2017#12]
I found there seems a typo [there](https://github.com/lfz/DSB2017/blob/master/net_detector.py#L99)

```python
        self.output = nn.Sequential(nn.Conv3d(self.featureNum_back[0], 64, kernel_size = 1),
                                    nn.ReLU(),
                                    #nn.Dropout3d(p = 0.3),
                                   nn.Conv3d(64, 5 * len(config['anchors']), kernel_size = 1))
```

To my limited knownledge, 1 out of the 5-dimension output should be the probability of a region proposal, which should be in [0,1], ususally we use sigmoid. Other 4 should be the bbox regressor, the linear activation is suitable. 

So I'm wondering whether it's just typo, or your explicit design?

**Comment by q5390498** [comment]
我还没有到这一步呢。请问你的数据处理完之后size都是[1,128.128,128]吗？为什么我的出来大多是128*128*128的，但有一些不是呢？

**Comment by duducheng** [comment]
我没有运行lfz的代码。如果你是完整且正确的运行他的代码，我觉得应该不会有问题，毕竟他的代码风格很清楚。如果你是自己处理的，那我认为是你可能没有做padding。他是取128的patch，所以需要在原始voxel上padding一圈64的。个人意见。

**Comment by q5390498** [comment]
我是直接运行lfz的代码的，只替换了数据，只有mhd的没有dicom的。就是detector/data.py里调用corp之后数据的size不统一，有大有小。

**Comment by lfz** [comment]
it's the logit before softmax

the softmax function is put in the loss.py

**Comment by duducheng** [comment]
Got it, thx.

To whom it may also concern, it should be [this line](https://github.com/lfz/DSB2017/blob/master/layers.py#L185)
